### PR TITLE
modules/../kratos: Initial commit

### PR DESCRIPTION
**Pull request by Kreyren**
Define bare minimum module that only deploys kratos for further contributions

###### Motivation for this change
Further development so that i am not relying on KeyCloak that takes excessive amount of resources to handle the Single Sign-On.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [-] aarch64-linux
  - [-] x86_64-darwin
  - [-] aarch64-darwin
- [-] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [-] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [-] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [-] (Package updates) Added a release notes entry if the change is major or breaking
  - [-] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kreyren** [comment]
>  (Module addition) Added a release notes entry if adding a new NixOS module

I don't know where are the relevant release notes as `doc/release-notes.xml` have latest entry in 2013

**Comment by bobby285271** [comment]
> I don't know where are the relevant release notes as `doc/release-notes.xml` have latest entry in 2013

[`nixos/doc/manual/release-notes/rl-2111.section.md`](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2111.section.md).

The Ci suggested that spaces are preferred for indents (and not TABs), would you like to change that?

**Comment by Kreyren** [comment]
Indentation changed on 4 spaces - https://github.com/Kreyren/nixpkgs/commit/d69697f37682871219de296fc948290dbd3a4cae

Release note added - [`58b65b4` (#137164)](https://github.com/NixOS/nixpkgs/pull/137164/commits/58b65b4b3efb61f6fbd4b7ee047705da1585174c)

Requested script ran - [`d29731e` (#137164)](https://github.com/NixOS/nixpkgs/pull/137164/commits/d29731e2cd29832b9f441d220bcd5cd42a74a7dc)

```console
[kreyren@leonid:~/Repositories/nixpkgs]$ nix-build nixos/release.nix -A manual.x86_64-linux
...

manual-combined.xml validates

man-pages-combined.xml validates
building '/nix/store/985q120wrrkqbg1dci5z8ax37zlsmdwa-manual-olinkdb.drv'...
Writing /nix/store/agywfg45f8mlvbnp8zc00rwxbrdg5gak-manual-olinkdb/manual.db for book(book-nixos-manual)
building '/nix/store/bbffhadj5z0d86sfvddpcg7rp7lvbyzy-nixos-manual-html.drv'...
Writing options.html for appendix(ch-options)
Writing release-notes.html for appendix(ch-release-notes)
Writing index.html for book(book-nixos-manual)
/nix/store/x4w6v9hi8b700zrkhmhr9ifsmsx2wyjv-nixos-manual-html

[kreyren@leonid:~/Repositories/nixpkgs]$ echo $?
0
```

**Comment by Kreyren** [comment]
> Checking EditorConfig / tests (pull_request_target) Failing after 33s — tests 

Error is not helpful:

image

**Comment by bobby285271** [comment]
> Indentation changed on 4 spaces

2 spaces is preferred (sorry for asking you to change this again).

> Error is not helpful:

That's weird, I can see this.

```
nixos/doc/manual/release-notes/rl-2111.section.md:
	Wrong line endings or new final newline

1 errors found
```

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 1]
```diff
@@ -320,3 +320,5 @@ To be able to access the web UI this port needs to be opened in the firewall.
 - GNOME desktop environment now enables `QGnomePlatform` as the Qt platform theme, which should avoid crashes when opening file chooser dialogs in Qt apps by using XDG desktop portal. Additionally, it will make the apps fit better visually.
 
 - `rofi` has been updated from '1.6.1' to '1.7.0', one important thing is the removal of the old xresources based configuration setup. Read more [in rofi's changelog](https://github.com/davatorium/rofi/blob/cb12e6fc058f4a0f4f/Changelog#L1).
+
+- Bare minimum module for `kratos` have been defined
```
Comment (bobby285271) [inline nixos/doc/manual/release-notes/rl-2111.section.md]: Nit:

```suggestion
- Bare minimum module for `kratos` have been defined.
```

File: nixos/modules/services/web-apps/kratos.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    services.kratos = {
+      enable = lib.mkEnableOption "Kratos, Libre and open source identity server";
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.kratos;
+        defaultText = "pkgs.kratos";
+        description = ''
+          The kratos package to use
+        '';
```
Comment (bobby285271) [inline nixos/modules/services/web-apps/kratos.nix:25]: Nit:

```suggestion
        description = "The kratos package to use.";
```

File: nixos/modules/services/web-apps/kratos.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      };
+    };
+  };
+
+  # config = lib.mkIf cfg.enable {
+  # };
+
+  meta.maintainers = [ ];
```
Comment (bobby285271) [inline nixos/modules/services/web-apps/kratos.nix:33]: Would you like to add yourself?

**Comment by Kreyren** [comment]
Indent changed on 2 spaces - [`bd60117` (#137164)](https://github.com/NixOS/nixpkgs/pull/137164/commits/bd60117cee74f251dbb4a02555b892166b2ab4b2)

Empty line added - [`fd068c5` (#137164)](https://github.com/NixOS/nixpkgs/pull/137164/commits/fd068c5f6ec0d0da9c225e7cf7f1942adcd2b9b8)

File: nixos/modules/services/web-apps/kratos.nix (hunk 3 above)
Comment (Kreyren) [reply nixos/modules/services/web-apps/kratos.nix:31]: Not at this time

File: nixos/doc/manual/release-notes/rl-2111.section.md (hunk 1 above)
Comment (Kreyren) [reply nixos/doc/manual/release-notes/rl-2111.section.md]: Unable to commit in GithubUI -> Commited in [`6b52f53` (#137164)](https://github.com/NixOS/nixpkgs/pull/137164/commits/6b52f5333a2ffe5aff32b90e05e1d3d3c5688f90)

**Comment by bobby285271** [comment]
And will be nice to squash all the changes.

**Comment by Kreyren** [comment]
> And will be nice to squash all the changes.

Can you squash using github so that the changes are archived since i reference them in comments?

**Comment by Kreyren** [comment]
I don't understand the failure in https://github.com/NixOS/nixpkgs/runs/3554458172

**Comment by bobby285271** [comment]
Maybe run `./nixos/doc/manual/md-to-db.sh` every time you update the release notes.

File: nixos/modules/services/web-apps/kratos.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        type = lib.types.package;
+        default = pkgs.kratos;
+        defaultText = "pkgs.kratos";
+        description = "The kratos package to use.";
+      };
+    };
+  };
+
+  # config = lib.mkIf cfg.enable {
+  # };
```
Comment (bobby285271) [inline nixos/modules/services/web-apps/kratos.nix:29]: I am not sure, but I was not able to find the implementation.

Comment (Kreyren) [reply nixos/modules/services/web-apps/kratos.nix:29]: elaborate "I was not able to find the implementation.". I though that the current is enough to deploy kratos through `services.kratos.enable` ?

Comment (bobby285271) [reply nixos/modules/services/web-apps/kratos.nix:29]: Sorry I am not familiar with this package, can you briefly describe how we should deploy this without this option? For example, adding the package to `environment.systemPackages` and maybe `systemd.packages`?

**Comment by Artturin** [comment]
> > And will be nice to squash all the changes.
> 
> Can you squash using github so that the changes are archived since i reference them in comments?

there's no need to have every single edit in their own commit nor is there a need to reference all commits in comments

**Comment by Artturin** [comment]
commit name should be nixos/kratos https://github.com/NixOS/nixpkgs/blob/5afb8c1aa7b48254b44fa5bc2cffc10943510e51/CONTRIBUTING.md#submitting-changes

**Review by samueldr — CHANGES_REQUESTED** [review]
This cannot be accepted, as it provides a broken unfinished module. Only "unplumbed" options.

**Pull request closed without merging**